lib/modules/noParticipation.js

```diff
@@ -108,9 +112,9 @@ function isSubscriber() {
 	return document.body.classList.contains('subscriber');
 }
 
-function nonNpLocation() {
-	const nonNpUrl = new URL(location.href);
-	nonNpUrl.hostname = 'www.reddit.com';
+export function nonNpLocation(href: string = location.href) {
+	const nonNpUrl = new URL(href, location.origin);
```

Comment:
Is `location.origin` supported on Edge? The [MDN page](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/origin#Browser_compatibility) says that compatibility info is unknown.

Reply:
The page linked above is about a global 'origin' property (window.origin)

window.location.origin [seems](https://developer.mozilla.org/en-US/docs/Web/API/Window/location) to be supported in all browsers except Opera

Reply:
Opera is based on Chromium, so generally will code which works on Chrome also work on Opera. In this use case it does not seem to matter whether `origin` or `href` is used, though.

Reply:
It was interesting to see though that location.origin always returns undefined in Opera. Leaving the PR as-is would have broken go-mode for our Opera users.

I'll be more careful about cross-browser incompatibility issues for future PRs. Thanks to mc10 for raising the issue.